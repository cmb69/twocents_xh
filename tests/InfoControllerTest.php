<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Twocents_XH.
 *
 * Twocents_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Twocents_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Twocents_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Twocents;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Twocents\Infra\FakeDb;
use Twocents\Infra\FakeRequest;
use Twocents\Infra\FakeSystemChecker;
use Twocents\Infra\View;

class InfoControllerTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $sut = new InfoController(
            "./plugins/twocents/",
            new FakeSystemChecker,
            new FakeDb,
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["twocents"])
        );
        $response = $sut(new FakeRequest());
        $this->assertEquals("Twocents 1.0", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
