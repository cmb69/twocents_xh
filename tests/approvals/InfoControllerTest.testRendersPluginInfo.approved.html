<!-- twocents plugin info -->
<h1>Twocents 1.2-dev</h1>
<div>
  <h2>System check</h2>
  <p class="xh_fail">PHP version is at least 7.1.0: failure</li>
  <p class="xh_fail">CMSimple_XH version is at least 1.7.0: failure</li>
  <p class="xh_fail">Plib_XH version is at least 1.4: failure</li>
  <p class="xh_warning">'db_folder' is writable: warning</li>
  <p class="xh_warning">'./plugins/twocents/config/' is writable: warning</li>
  <p class="xh_warning">'./plugins/twocents/css/' is writable: warning</li>
  <p class="xh_warning">'./plugins/twocents/languages/' is writable: warning</li>
</div>
